flake.nix

```diff
@@ -0,0 +1,58 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
```

Comment:
can we please try to stay on stable?

Reply:
> That sounds like the way to go then? 

Yup, working on it :)

>The alternative would be abandoning caching, which sounds like it would make testing the nix components non-viable long term (we shouldn't have to wait like 15' each build, that _has to_ get better asap).

Currently the benefit is we have a working flake for minimint via crane. Also looking at caching on crane, so we get the best of it, till we get cargo2nix working